### PR TITLE
resolve #9500

### LINKED ISSUE #9500: P- should remove whole project directory, not only rc file
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Arch Linux x86 64
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            |x86/64
| r2 -v full output, **not truncated** (mandatory)     | radare2 2.4.0-git 17397 @ linux-x86-64 git.2.2.0-583-gd225c83d1 commit: d225c83d1164581375b4340b169d85f30b9db964 build: 2018-02-24__19:03:01

### Expected behavior
The whole directory `~/.config/radare2/projects/qwerty` should be removed.

### Actual behavior
Only `~/.config/radare2/projects/qwerty/rc` file is removed.

### Steps to reproduce the behavior 
```
$ r2 -
[0x00000000]> Ps qwerty
qwerty
[0x00000000]> P- qwerty
rm /home/g/.config/radare2/projects/qwerty/rc
[0x00000000]> !ls -la /home/g/.config/radare2/projects|grep qwerty
drwxr-xr-x 2 g g 4096 Feb 24 19:04 qwerty
```

### PR DESCRIPTION
**Pull request by jbhv12**

### CODE REVIEW

File: libr/core/project.c [hunk 1]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		//rm xrefs.sdb file
+		path = r_file_dirname (path);
+		path = r_str_append (path, R_SYS_DIR);
+		path = r_str_append (path, "xrefs.sdb");
+		if(r_file_exists (path)) {
+			r_file_rm (path);
+			eprintf ("rm %s\n", path);
+		}
+		//change path to rop.d
+		path = r_file_dirname (path);
```
Comment (radare) [inline libr/core/project.c]: Memory leak

File: libr/core/project.c [hunk 2]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		path = r_str_append(path,R_SYS_DIR);
```
Comment (radare) [inline libr/core/project.c]: Missing space befor (

File: libr/core/project.c [hunk 3]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		path = r_str_append(path,"rop.d");
```
Comment (radare) [inline libr/core/project.c]: You can strappend once rsysdir is a constant just concat them

File: libr/core/project.c [hunk 4]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
-		eprintf ("rm %s\n", path);
-		path = r_str_append (path, ".d");
+		if(r_file_exists (path)) {
+			r_file_rm (path);
+			eprintf ("rm %s\n", path);
+		}
+
+		//rm xrefs.sdb file
+		path = r_file_dirname (path);
+		path = r_str_append (path, R_SYS_DIR);
```
Comment (radare) [inline libr/core/project.c]: Dont append constants twice

File: libr/core/project.c [hunk 5]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		path = r_str_append (path, "xrefs.sdb");
+		if(r_file_exists (path)) {
```
Comment (radare) [inline libr/core/project.c]: Missing space befoee (

File: libr/core/project.c [hunk 6]
```diff
@@ -176,9 +176,24 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
 	}
 	if (r_core_is_project (core, prjfile)) {
 		// rm project file
-		r_file_rm (path);
-		eprintf ("rm %s\n", path);
-		path = r_str_append (path, ".d");
```
Comment (radare) [inline libr/core/project.c]: Missing space before (

**Comment by radare** [comment]
ping

**Comment by jbhv12** [comment]
Sorry for delay. On it.

File: libr/core/project.c (hunk 1 above)
Comment (jbhv12) [reply libr/core/project.c]: Why would there be memory leak? `path` variable is freed at the end of the function. Am I missing something?

**Comment by radare** [comment]
Read the code if rfiledirname

File: libr/core/project.c [hunk 7]
```diff
@@ -175,31 +176,57 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+			r_file_rm (path);
+			eprintf ("rm %s\n", path);
+		}
+
+		prjDir = r_file_dirname (path);
+		free (path);
+
+		//rm xrefs.sdb file
+		length = strlen (prjDir) + strlen (R_SYS_DIR) + strlen ("xrefs.sdb") + 1;
+		tmp = (char *)malloc (length);
```
Comment (radare) [inline libr/core/project.c]: Use r_str_newf and save 4 lines

Comment (radare) [reply libr/core/project.c]: You can also use sdb_fmt and avoid the free

File: libr/core/project.c [hunk 8]
```diff
@@ -175,31 +176,57 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		strcat (tmp, "xrefs.sdb");
+		if (r_file_exists (tmp)) {
+			r_file_rm (tmp);
+			eprintf ("rm %s\n", tmp);
+		}
+		free (tmp);
+
+		//change path to rop.d
+		length = strlen (prjDir) + strlen (R_SYS_DIR) + strlen ("rop.d") + 1;
+		tmp = (char *)malloc (length);
```
Comment (radare) [inline libr/core/project.c]: Use rstrnewf or sdb_fmt

**Comment by radare** [comment]
btw the current way to use P- is by prefixing it with a dot to run the commands that are printed.

**Comment by jbhv12** [comment]
@radare Can you help me out? I can't figure out what's wrong here. why does git clone exit with 128?

**Comment by giuscri** [comment]
@jbhv12 that has been fixed. Try to make travis to re-run by pushing a new commit.

File: libr/core/project.c [hunk 9]
```diff
@@ -164,7 +164,8 @@ R_API int r_core_project_list(RCore *core, int mode) {
 }
 
 R_API int r_core_project_delete(RCore *core, const char *prjfile) {
-	char *path;
+	char *path, *tmp, *prjDir;
```
Comment (giuscri) [inline libr/core/project.c]: camelCase is not used a lot in the codebase

File: libr/core/project.c [hunk 10]
```diff
@@ -175,31 +176,49 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
 		return false;
 	}
 	if (r_core_is_project (core, prjfile)) {
+
```
Comment (giuscri) [inline libr/core/project.c]: ?

**Comment by giuscri** [comment]
you should fork radare2-regressions and change `BROKEN=1` at https://github.com/radare/radare2-regressions/blob/master/t/cmd_project#L282-L331

> When relevant, write a test for radare2-regressions and submit a PR also there. Use the same branch name in both repositories, so Travis will be able to use your new tests together with new changes. AppVeyor (for now) still uses radare/radare2-regressions repo with branch master. NOTE: when merging PRs, always merge the radare2-regressions PR first.

File: libr/core/project.c [hunk 11]
```diff
@@ -175,31 +176,49 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		if (r_file_exists (path)) {
+			r_file_rm (path);
+			eprintf ("rm %s\n", path);
+		}
+
+		prjDir = r_file_dirname (path);
+		free (path);
+
+		//rm xrefs.sdb file
+		tmp = r_str_newf ("%s%s%s", prjDir, R_SYS_DIR, "xrefs.sdb");
```
Comment (giuscri) [inline libr/core/project.c]: why `tmp`? imo `xrefs` is a better name

File: libr/core/project.c [hunk 12]
```diff
@@ -175,31 +176,49 @@ R_API int r_core_project_delete(RCore *core, const char *prjfile) {
...
+		if (r_file_exists (tmp)) {
+			r_file_rm (tmp);
+			eprintf ("rm %s\n", tmp);
+		}
+		free (tmp);
+
+		//change path to rop.d
+		tmp = r_str_newf ("%s%s%s", prjDir, R_SYS_DIR, "rop.d");
```
Comment (giuscri) [inline libr/core/project.c]: same as above. don't think `tmp` is a good name

**Comment by giuscri** [comment]
> btw the current way to use P- is by prefixing it with a dot to run the commands that are printed.

@radare `rmdir` is actually called https://github.com/radare/radare2/blob/master/libr/core/project.c#L179

File: libr/core/project.c (hunk 9 above)
Comment (jbhv12) [reply libr/core/project.c]: I used it because i saw it in other functions in the same file. Should I change it to something like `prj_dir`?

Comment (giuscri) [reply libr/core/project.c]: ok, you're right then sorry

**Comment by giuscri** [comment]
to help travis not failing do a
```
git remote upstream add git@github.com:radare/radare2.git
git fetch upstream
git rebase master
git push origin master -f
```

**Pull request merged**